tools/crm_resource.c

```diff
@@ -779,30 +781,15 @@ main(int argc, char **argv)
         CMD_ERR("Resource '%s' not found: %s", crm_str(rsc_id), pcmk_strerror(rc));
 
     } else if (rsc_cmd == 'W') {
-        if (rsc_id == NULL) {
-            CMD_ERR("Must supply a resource id with -r");
-            rc = -ENXIO;
-            goto bail;
-        }
         rc = cli_resource_search(rsc_id, &data_set);
         if (rc >= 0) {
             rc = pcmk_ok;
         }
 
     } else if (rsc_cmd == 'q') {
-        if (rsc_id == NULL) {
-            CMD_ERR("Must supply a resource id with -r");
-            rc = -ENXIO;
-            goto bail;
-        }
         rc = cli_resource_print(rsc_id, &data_set, TRUE);
 
     } else if (rsc_cmd == 'w') {
-        if (rsc_id == NULL) {
-            CMD_ERR("Must supply a resource id with -r");
-            rc = -ENXIO;
-            goto bail;
-        }
         rc = cli_resource_print(rsc_id, &data_set, FALSE);
 
     } else if(rsc_cmd == 'Y') {
```

Comment:
Oops, we missed some options that *don't* require a resource when we set require_resource. 'Y', 'O', and 'o' can be used with or without specifying a resource.

Reply:
Nice Catch Ken Thanks alot for finding this bug :)